### ISSUE TITLE
Add `table` plugin to @gravity-ui/chartkit

### ISSUE THREAD

**Issue by korvin89** [gravity-ui/rfc#8]
## Objective

Hi there! I'm from [DataLens](https://datalens.tech/) team and now we are using [@gravity-ui/react-data-table](https://github.com/gravity-ui/react-data-table) solution to render tables in our application. It is a pretty powerful component but with restrictions in architecture for extending. We have a lack of core features like `horizontal virtualization`, `sticky columns` and convenient `css api`. Given the current architecture of the component, these improvements will require a complete rethink of the current code. This table is successfully used in many large applications, its code base is completely not covered by tests, so I consider it advisable not to make such serious changes to it.

## Solution Proposal

I propose to organize the `table` plugin in [chartkit](https://github.com/gravity-ui/chartkit) lib based on [tanstack](https://tanstack.com/table/latest/docs/introduction) table.

#### Why tanstack and not [uikit table](https://github.com/gravity-ui/uikit/tree/main/src/components/Table)?
Uikit table is convenient for displaying a small amount of content in accordance with the [Gravity UI design system](https://gravity-ui.com/design/branding/resources), but it has big performance problems and also does not have the necessary functionality described above.

## Definition of done

[Chartkit](https://github.com/gravity-ui/chartkit) lib should have the new plugin `table` for table visualisation.

**Comment by korvin89** [comment]
cc @kuzmadom @resure

**Comment by ma-efremoff** [comment]
`sticky rows` feature is present in `@gravity-ui/react-data-table` (`headerData`/`footerData`)

**Comment by korvin89** [comment]
> `sticky rows` feature is present in `@gravity-ui/react-data-table` (`headerData`/`footerData`)

Are there any examples? It is a little bit unexpected behaviour from properties with such names :)

**Comment by ma-efremoff** [comment]
> Are there any examples?

https://preview.gravity-ui.com/react-data-table/?path=/story/react-data-table--default&args=fixHead:!true;fixFooter:!true

> It is a little bit unexpected behaviour from properties with such names :)

It is a good time (cause of uikit6) to rename them, so if you know better names please file an issue to the repo.

Anyway, `sticky rows` is not the only feature missing in react-data-table.

**Comment by korvin89** [comment]
> > Are there any examples?
> 
> https://preview.gravity-ui.com/react-data-table/?path=/story/react-data-table--default&args=fixHead:!true;fixFooter:!true
> 
> > It is a little bit unexpected behaviour from properties with such names :)
> 
> It is a good time (cause of uikit6) to rename them, so if you know better names please file an issue to the repo.
> 
> Anyway, `sticky rows` is not the only feature missing in react-data-table.

Wow! Thank you. I've fixed description.

**Comment by refreakk** [comment]
Hello everyone! I might be off-topic, but perhaps it's time to update the [@gravity-ui/react-data-table](https://github.com/gravity-ui/react-data-table) library to the new version of React? I really like the implementation of this particular table, but I can't use it in my projects.

I wouldn't mind participating in the update to the new version :)

**Comment by korvin89** [comment]
> Hello everyone! I might be off-topic, but perhaps it's time to update the [@gravity-ui/react-data-table](https://github.com/gravity-ui/react-data-table) library to the new version of React? I really like the implementation of this particular table, but I can't use it in my projects.
> 
> I wouldn't mind participating in the update to the new version :)

You cannot use table with react 18?

**Comment by refreakk** [comment]
> > Hello everyone! I might be off-topic, but perhaps it's time to update the [@gravity-ui/react-data-table](https://github.com/gravity-ui/react-data-table) library to the new version of React? I really like the implementation of this particular table, but I can't use it in my projects.
> > I wouldn't mind participating in the update to the new version :)
> 
> You cannot use table with react 18?

Yes, [@gravity-ui/react-data-table](https://github.com/gravity-ui/react-data-table) support react 16.14.0 and down

**Comment by korvin89** [comment]
> Yes, [@gravity-ui/react-data-table](https://github.com/gravity-ui/react-data-table) support react 16.14.0 and down

And above [>=16.14.0](https://github.com/gravity-ui/react-data-table/blob/main/package.json#L96C15-L96C24). So, there is no difficulties to use this table with react 17 | 18

**Comment by refreakk** [comment]
It really works. Thank you.

Two days ago it was showing errors on the React and React Dom versions, on vite.js.

**Comment by SeqviriouM** [comment]
@refreakk @korvin89 is this issue still relevant?

**Comment by korvin89** [comment]
> @refreakk @korvin89 is this issue still relevant?

Of course! It will take a lot of time I suppose

**Comment by resure** [comment]
@korvin89 should we mark it as `in progress` then?

**Comment by korvin89** [comment]
> @korvin89 should we mark it as `in progress` then?

Sure

**Comment by SeqviriouM** [comment]
@korvin89 can you share the updates?

**Comment by korvin89** [comment]
@SeqviriouM It's still in development
